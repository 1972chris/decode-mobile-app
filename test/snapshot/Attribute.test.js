import React from 'react';
import renderer from 'react-test-renderer';
import AttributeComponent from '../../application/components/Attribute/Attribute';

// Avoid warning related to Switch component
// https://github.com/facebook/react-native/issues/16247
jest.mock('Switch', () => 'Switch');

it('renders Attribute on verfied mode component', () => {
  const rendered = renderer.create(<AttributeComponent value="Barcelona" isVerified name="test" />).toJSON();
  expect(rendered).toMatchSnapshot();
});

it('renders Attribute on non-verfied mode component', () => {
  const rendered = renderer.create(<AttributeComponent value="Barcelona" isVerified={false} name="test" />).toJSON();
  expect(rendered).toMatchSnapshot();
});
