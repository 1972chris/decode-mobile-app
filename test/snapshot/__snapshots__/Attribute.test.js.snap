// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Attribute on non-verfied mode component 1`] = `
<View
  style={
    Object {
      "alignSelf": "stretch",
      "backgroundColor": "#FFF",
      "display": "flex",
      "flexDirection": "column",
      "marginVertical": 20,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "rgba(0, 0, 0, 0.87)",
          "fontSize": 16,
          "fontWeight": "500",
        }
      }
    >
      test
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#9B9B9B",
          "flex": 1,
          "fontSize": 14,
          "marginLeft": 5,
        }
      }
    >
      Barcelona
    </Text>
    <Switch
      onValueChange={[Function]}
      value={true}
    />
  </View>
</View>
`;

exports[`renders Attribute on verfied mode component 1`] = `
<View
  style={
    Object {
      "alignSelf": "stretch",
      "backgroundColor": "#FFF",
      "display": "flex",
      "flexDirection": "column",
      "marginVertical": 20,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "rgba(0, 0, 0, 0.87)",
          "fontSize": 16,
          "fontWeight": "500",
        }
      }
    >
      test
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#9B9B9B",
          "flex": 1,
          "fontSize": 14,
          "marginLeft": 5,
        }
      }
    >
      Barcelona
    </Text>
    <Switch
      onValueChange={[Function]}
      value={true}
    />
  </View>
</View>
`;
